src/reformat.c

```diff
@@ -536,7 +536,7 @@ static avifResult avifImageYUVAnyToRGBAnySlow(const avifImage * image,
     assert((alphaMultiplyMode == AVIF_ALPHA_MULTIPLY_MODE_NO_OP) || aPlane);
 
     for (uint32_t j = 0; j < image->height; ++j) {
-        const uint32_t uvJ = j >> state->formatInfo.chromaShiftY;
+        const uint32_t uvJ = j >> (state->formatInfo.monochrome ? 0 : state->formatInfo.chromaShiftY);
```

Comment:
Nit: This change is not necessary because `uvJ` is only used when `hasColor` is true.

If you prefer a change, then it'd be better to use `hasColor` to match lines 567-568.